src/attr/_make.py

```diff
@@ -1190,52 +1210,43 @@ def __ne__(self, other):
 
 
 def _make_eq(cls, attrs):
+    """
+    Create equality methods.
+    """
     attrs = [a for a in attrs if a.eq]
 
-    unique_filename = _generate_unique_filename(cls, "eq")
-    lines = [
-        "def __eq__(self, other):",
-        "    if other.__class__ is not self.__class__:",
-        "        return NotImplemented",
-    ]
+    def attrs_to_tuple(obj):
+        """
+        Save us some typing.
+        """
+        return _attrs_to_cmp_tuple(obj, attrs)
+
     # We can't just do a big self.x = other.x and... clause due to
     # irregularities like nan == nan is false but (nan,) == (nan,) is true.
-    if attrs:
-        lines.append("    return  (")
-        others = ["    ) == ("]
-        for a in attrs:
-            lines.append("        self.%s," % (a.name,))
```

Comment:
I think we should strive to keep this generated code behavior here if possible. It looks like a bit of work and tuning was put into getting a bunch of details right (for speed, debugging, serialization), and I don't want to throw that out to solve and edgier edge case. I have an idea, i'll try it in my coming PR